index.bs

```diff
@@ -1,11 +1,205 @@
-<pre class='metadata'>
-Title: Web Extensions
-Shortname: webextensions
-Level: 1
-Status: w3c/CG-DRAFT
-Group: WECG
-URL: https://w3c.github.io/webextensions
-Editor: Mukul Purohit, Microsoft Corporation https://www.microsoft.com, mpurohit@microsoft.com
-Editor: Tomislav Jovanovic, Mozilla https://www.mozilla.org/, tjovanovic@mozilla.com
-Abstract: [Placeholder] Abstract.
-</pre>
+<pre class='metadata'>
+Title: Web Extensions
+Shortname: webextensions
+Level: 1
+Status: w3c/CG-DRAFT
+Group: WECG
+URL: https://w3c.github.io/webextensions
+Editor: Mukul Purohit, Microsoft Corporation https://www.microsoft.com, mpurohit@microsoft.com
+Editor: Tomislav Jovanovic, Mozilla https://www.mozilla.org/, tjovanovic@mozilla.com
+Abstract: [Placeholder] Abstract.
+Markup Shorthands: markdown yes
+</pre>
+
+# Manifest
+
+A WebExtension must have a manifest file at its root directory.
+
+## Manifest file
+
+A manifest file is a [[!JSON]] document named `manifest.json`. Malformed JSON files are not supported.
+
+## Manifest keys
+
+If manifest keys that are not defined in this specification are specified, implementors must ignore those keys.
+
+If manifest keys that are defined in this specification are specified with a different JSON type than defined in this specification, implementors must ignore those keys.
+
+The following keys must be considered valid:
+
+* <a href="#key-manifest_version">`manifest_version`</a>: required.
+* <a href="#key-name">`name`</a>: required.
+* <a href="#key-version">`version`</a>: required.
+* <a href="#key-default_locale">`default_locale`</a>: required under some conditions.
+* <a href="#key-description">`description`</a>: optional.
+* <a href="#key-icons">`description`</a>: optional.
+* <a href="#key-author">`author`</a>: optional
+* <a href="#key-background">`background`</a>: optional
+* <a href="#key-browser_action">`browser_action`</a>: optional
```

Comment:
This raises the question, should we document `browser_action` and `page_action` or document the Manifest v3 `action`?

Reply:
I don’t know Manifest v3 very much so I just put what was existing on several browsers right now. This should indeed be properly discussed…

Reply:
Created https://github.com/w3c/webextensions/issues/50 for discussion

Reply:
I synced with my team on this specific question and the border issue of how we're thinking about Manifest V2 and Manifest V3 in the context of this spec. I'll be following up in #50 shortly.

Reply:
done!

Reply:
Should I add `host_permissions` as well then? Since it’s the other manifest key that’s not in manifest v2, as far as I know.